Add readonly mode to APY daemon

ruuda opened the pull request:
I wanted to test some changes in the server part, but have it run against an existing database that a different instance of the daemon is already writing to. So add a read-only mode (`--fetch disable`) that runs only the http server part and not the fetch-prices-from-the-chain part.

I would have liked to have `--fetch` and `--no-fetch` flags instead, but I don’t know how to do this with Clap `#[derive]` macros. I found some discussions about it with workarounds that use the method-chaining builders. So for now let’s make it an option instead.

enriquefynn commented:
We could use a boolean flag `--read-only`.
```rust
#[clap(long)]
read_only: bool,
```
What do you think?

ruuda commented:
Yeah, that’s nicer, I’ll do that.

ruuda commented:
I switched the flag. I also noticed that the `/apy` endpoint returned a misleading error, I refactored the route parsing to return more helpful errors, and I think it’s a bit simpler like this as well. Please take another look.

enriquefynn commented on cli/listener/src/main.rs:
```diff
@@ -529,41 +536,37 @@ fn get_interval_price_request(
-        "metrics" => Some(ParseUrlResponse::Metrics),
-        _ => None,
+        Some("metrics") => Ok(Endpoint::Metrics),
+        _ => Err(ResponseError::NotFound("Unknown route.")),
```
👍

enriquefynn reviewed (approved):
LGTM!

ruuda commented on cli/listener/src/main.rs:
```diff
@@ -529,41 +536,37 @@ fn get_interval_price_request(
-            };
-            Some(ParseUrlResponse::IntervalPriceRequest(dates))
+    let base = Url::parse("http://unused.invalid/").expect("Hard-coded value is valid.");
+    let parse_result = Url::options().base_url(Some(&base)).parse(request_url);
```
Not sure if it makes a difference in practice, but it looked like a more "proper" way to do it from the docs.
